source/core/solid_base.cc

```diff
@@ -104,12 +190,16 @@ SolidBase<dim, spacedim>::setup_triangulation(const bool restart)
   // Refine the solid triangulation to its initial size
   // NB: solid_tria should not be refined if loaded from a restart file
   // afterwards
-  if (!restart)
+  if (!restart && !param->solid_mesh.simplex)
     {
       solid_tria->refine_global(param->solid_mesh.initial_refinement);
       // Initialize dof handler for solid
-      FE_Q<dim, spacedim> fe(1);
-      solid_dh.distribute_dofs(fe);
+      solid_dh.distribute_dofs(*fe);
+    }
+  else if (!restart)
+    {
+      // Initialize dof handler for solid
```

Comment:
Don't you need to globally refine the triangulation in this case as well?

Reply:
With the simplex triangulation it is not possible for now, but I can do it when/if it is implemented in dealii